Email Editor - WP 6.5 compatibility fixes

costasovo opened the pull request:
## Description

This PR fixes issues found when we tested the plugin with WP 6.5 beta 1. Some the issues in the ticket were addressed in different PRs, but I've also found a couple of new issues.

#### Issues addressed in this PR:

1. The appearance of the toolbar menu
2. Sidebar header (tabs)
3. The main area appearance
4. Inserter goes below the browser window
5. Blocks supporting borders allow shadow configuration - I disabled this for now.
6. Fixed Toolbar was not working

#### Issues I was not able to replicate (I was working with beta 2 so possibly some issues were already fixed)

1. Missing Image settings
2. Image missing in Preview
3. The Button's full-width setting is not working (possibly related to 3) The main area appearance)
4. Column block spacing is not working (possibly related to 3) The main area appearance)

#### Issues solved in different PR/ticket

1. Padding cannot be changed (https://github.com/mailpoet/mailpoet/pull/5450)
2. Button rendering is not working (https://github.com/mailpoet/mailpoet/pull/5450)

## Code review notes

I was not able to replicate all the issues (e.g. the problems with image blocks) and also found that some of those in the ticket were not related to WP 6.5 but they were just issues in the rederer (e.g. problems with padding in button were addressed in https://github.com/mailpoet/mailpoet/pull/5450). 

I fixed the issues with BC in mind so the editor should work in WP 6.4 and WP 6.5.

Two issues worth additional comments are:

1. Problem with Tabs in Sidebar Header - The post editor switched to [Tabs component](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/tabs) but this one is in private APIs and not available in WP 6.4. I was able to make it work for WP 6.5 but since it is still private and not supported in WP6.4 I chose to fix the issue by adding CSS for the current implemantation.
2. Fixed Toolbar Not Working - The fix for WP 6.5 caused that in WP 6.4 we rendered two toolbars in the header. I needed to add a condition based on WP/Gutenberg version. I'm not happy with such a condition, but I hope we can get rid of it as soon as we drop support of WP 6.4.

## QA notes

Please test the editor in WP 6.4 and WP 6.5 Beta. There might be some differences because WP6.5 provides newer Gutenberg packages, but we should not introduce any regression compared to WP 6.4

Veljko's note to Alex: We can use site cloning on InstaWP or creating a new site instance to test the difference.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5907]

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-5907]: https://mailpoet.atlassian.net/browse/MAILPOET-5907?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

costasovo committed:
Remove unnecessary className

The class name is added automatically. Adding it explicitly causes a duplicity.
[MAILPOET-5907]

costasovo committed:
Fix the left header toolbar appearance

[MAILPOET-5907]

costasovo committed:
Fix desktop email content view

The condition stopped working because useResizeCanvas started returning
some styles also for Desktop
[MAILPOET-5907]

costasovo committed:
Fix sidebar tabs appearance for WP 6.5

I explored using the private Tabs component and it worked, but it
doesn't work with WP 6.4 so I went with adding CSS.
We can switch to Tabs later. Hopefully they won't be private.
[MAILPOET-5907]

costasovo committed:
Disable shadow for all blocks

The shadow is currently not opt-in and can't be disabled in theme.json
[MAILPOET-5907]

costasovo committed:
Add fixed block toolbar support for WP 6.5

The fixed toolbar used to be just an absolutely positioned toolbar.
After the recent changes, we need to render it ourselves.
[MAILPOET-5907]

costasovo committed:
Add fixed block toolbar collapse toggle

[MAILPOET-5907]

costasovo committed:
Add a workaround for fixed toolbar backward compatibility

In WP 6.4 and below is the fixed toolbar rendered within the same
slot as the floating toolbar but is positioned via CSS.

In WP 6.5, the fixed toolbar needs to be rendered in the Header explicitly.
This commit adds a workaround to backward compatibility with WP 6.4 to prevent rendering
two fixed toolbars.

Inspired by https://github.com/woocommerce/woocommerce/pull/43835/commits/1abf85fb8336a65ac66d1be9318ce5224ac84e12
[MAILPOET-5907]

costasovo committed:
Fix the padding issue in the preview dropdown menu

The preview in the new tab button had a different padding.
[MALPOET-5907]

costasovo committed:
Ensure correct display when switching mobile/desktop preview

useResizeCanvas hook started to return some values for margin, but only after the switch happened and
they were breaking margins
[MAILPOET-5907]

costasovo committed:
Fix border radius issues in the button

When only the radius was defined, we set border solid and it caused
that sometimes it was rendered even when the width was not defined.
Overflow hidden ensures that inner link element doesn't break the border radius.
[MAILPOET-5907

costasovo committed:
Disable unwanted focus outline on buttons in sidebar

[MAILPOET-5907]

lysyjan reviewed (dismissed):
LGTM

Aschepikov commented:
Hi @costasovo 
With the 4. Inserter goes below the browser window
I see that it is fixed in cases when the border of the window is very close to the block like here
![image](https://github.com/mailpoet/mailpoet/assets/113185437/3e33b184-6502-4c56-8c0e-b645efb6de41)
But in the cases when the border is not so close it is still going below the window (yep there is a scrollbar but it is hard to select anything)
![image](https://github.com/mailpoet/mailpoet/assets/113185437/85fe8ff9-0a8b-4611-82f6-4e30e047a25d)

The border in the preview and the emails is always black even when it is not selected, white, or with another color
![image](https://github.com/mailpoet/mailpoet/assets/113185437/f80ec088-6679-4cc2-9728-a4a0e7ab6a59)

Aschepikov commented:
Also there are Some conflicts can you resolve them pls @costasovo

costasovo committed:
Use a smaller bottom margin for the canvas to prevent hidden inserter popup

[MAILPOET-5907]

costasovo commented:
@Aschepikov Thanks for the detailed checking. I changed the bottom padding of the canvas to mitigate the issue with the hidden inserter. The issue with the border colors was fixed in a different PR. I updated this one on top of the trunk, so border colors should work as well now.

@lysyjan I rebased the branch on the trunk and added one new commit. Can you please take a look?

veljkho reviewed (approved):
I see some issues still but will be testing against trunk and open a separate Jira ticket to solve them.
